### PR TITLE
[TGUI] Crafting

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Converts crafting from NanoUI to TGUI

## Why It's Good For The Game
Nano you have 4 interfaces left. You will not cling to this world much longer.

## Images of changes
Expanded mode
image

Expanded mode, craftable only
image

Compact mode
image

Compact mode, craftable only
image

Crafting
image

## Changelog
:cl: AffectedArc07
add: Personal crafting now uses TGUI
/:cl:

### CODE REVIEW

**Comment by ppi13** [comment]
Death to NanoUI!

File: code/modules/crafting/craft.dm [hunk 1]
```diff
@@ -341,49 +344,50 @@
...
-	if(href_list["toggle_compact"])
-		display_compact = !display_compact
-		. = TRUE
-
-	SSnanoui.update_uis(src)
+	. = TRUE
+
+	switch(action)
+		if("make")
+			var/datum/crafting_recipe/TR = locate(params["make"]) in GLOB.crafting_recipes
```
Comment (dearmochi) [inline code/modules/crafting/craft.dm:360]: What happens if `params["make"]` refers to null?

File: code/modules/crafting/craft.dm [hunk 2]
```diff
@@ -341,49 +344,50 @@
...
+			busy = TRUE
+			SStgui.update_uis(src)
+			var/fail_msg = construct_item(usr, TR)
+			if(!fail_msg)
+				to_chat(usr, "<span class='notice'>[TR.name] constructed.</span>")
+				if(TR.alert_admins_on_craft)
+					message_admins("[key_name_admin(usr)] has created a [TR.name] at [ADMIN_COORDJMP(usr)]")
+			else
+				to_chat(usr, "<span class ='warning'>Construction failed[fail_msg]</span>")
```
Comment (dearmochi) [inline code/modules/crafting/craft.dm:369]:
```suggestion
				to_chat(usr, "<span class='warning'>Construction failed[fail_msg]</span>")
```

**Review by dearmochi — APPROVED** [review]
Looks good

**Pull request merged**